src/daemon/daemon_rpc.cpp

```diff
@@ -267,7 +267,18 @@ grpc::Status mp::DaemonRpc::authenticate(grpc::ServerContext* context, const Aut
 
     if (status.ok())
     {
-        accept_cert(client_cert_store, client_cert_from(context), server_address);
+        try
+        {
+            accept_cert(client_cert_store, client_cert_from(context), server_address);
+        }
+        catch (const mp::CertStoreException& e)
+        {
+            return grpc::Status{grpc::StatusCode::UNAUTHENTICATED, e.what()};
+        }
+        catch (const std::exception& e)
+        {
+            mpl::log(mpl::Level::warning, category, e.what());
+        }
```

Comment:
Shouldn't we let this through so it ends up as an error on the client?

Reply:
I have an idea I'll push that should alleviate my concern.

Reply:
Ok, lemme know what you think.